tests/lib/managers/webhooks-test.js

```diff
@@ -151,4 +152,73 @@ describe('Webhooks', function() {
+				'box-delivery-timestamp': '2020-01-01T00:00:00-07:00',
+				'box-signature-algorithm': 'HmacSHA256',
+				'box-signature-primary': '6TfeAW3A1PASkgboxxA5yqHNKOwFyMWuEXny/FPD5hI=',
+				'box-signature-secondary': 'v+1CD1Jdo3muIcbpv5lxxgPglOqMfsNHPV899xWYydo=',
+				'box-signature-version': '1'
+			},
+			PRIMARY_SIGNATURE_KEY = 'SamplePrimaryKey',
+			SECONDARY_SIGNATURE_KEY = 'SampleSecondaryKey',
+			INCORRECT_SIGNATURE_KEY = 'IncorrectKey',
+			DATE_IN_PAST = Date.parse('2010-01-01T00:00:00-07:00'),
+			DATE_IN_FUTURE = Date.parse('2030-01-01T00:00:00-07:00'),
+			HEADERS_WITH_WRONG_SIGNATURE_VERSION = Object.assign({}, HEADERS, {'box-signature-version': '2'}),
+			HEADERS_WITH_WRONG_SIGNATURE_ALGORITHM = Object.assign({}, HEADERS, {'box-signature-algorithm': 'XXX'});
+
+		it('should validate the webhook message with two good keys', function() {
+			const clock = sinon.useFakeTimers(DATE_IN_PAST);
+			assert.ok(Webhooks.validateMessage(BODY, HEADERS, PRIMARY_SIGNATURE_KEY, SECONDARY_SIGNATURE_KEY));
+			clock.restore();
+		});
+
+		it('should validate the webhook message with good primary key', function() {
+			const clock = sinon.useFakeTimers(DATE_IN_PAST);
+			assert.ok(Webhooks.validateMessage(BODY, HEADERS, PRIMARY_SIGNATURE_KEY, INCORRECT_SIGNATURE_KEY));
+			clock.restore();
+		});
+
+		it('should validate the webhook message with good secondary key', function() {
+			const clock = sinon.useFakeTimers(DATE_IN_PAST);
+			assert.ok(Webhooks.validateMessage(BODY, HEADERS, INCORRECT_SIGNATURE_KEY, SECONDARY_SIGNATURE_KEY));
+			clock.restore();
+		});
+
+		it('should NOT validate the webhook message with two wrong keys', function() {
+			const clock = sinon.useFakeTimers(DATE_IN_PAST);
+			assert.ok(!Webhooks.validateMessage(BODY, HEADERS, INCORRECT_SIGNATURE_KEY, INCORRECT_SIGNATURE_KEY));
+			clock.restore();
+		});
+
+		it('should NOT validate the webhook message with two null keys', function() {
+			const clock = sinon.useFakeTimers(DATE_IN_PAST);
+			assert.ok(!Webhooks.validateMessage(BODY, HEADERS, null, null));
+			clock.restore();
+		});
+
+		it('should NOT validate the webhook message if it is too old', function() {
+			const clock = sinon.useFakeTimers(DATE_IN_FUTURE);
+			assert.ok(!Webhooks.validateMessage(BODY, HEADERS, PRIMARY_SIGNATURE_KEY, SECONDARY_SIGNATURE_KEY));
+			clock.restore();
+		});
+
+		it('should NOT validate the webhook message if the signature version is wrong', function() {
+			const clock = sinon.useFakeTimers(DATE_IN_PAST);
+			assert.ok(!Webhooks.validateMessage(BODY, HEADERS_WITH_WRONG_SIGNATURE_VERSION, PRIMARY_SIGNATURE_KEY, SECONDARY_SIGNATURE_KEY));
+			clock.restore();
+		});
+
+		it('should NOT validate the webhook message if the signature algorithm is wrong', function() {
+			const clock = sinon.useFakeTimers(DATE_IN_PAST);
+			assert.ok(!Webhooks.validateMessage(BODY, HEADERS_WITH_WRONG_SIGNATURE_ALGORITHM, PRIMARY_SIGNATURE_KEY, SECONDARY_SIGNATURE_KEY));
+			clock.restore();
+		});
```

Comment:
Could we add a test to ensure that the passed-in `maxMessageAge` value is used when it's passed in?